apps/costcontrol/js/settings/limitdialog.js

```diff
@@ -76,7 +76,7 @@ function dataLimitConfigurer(guiWidget, settings, viewManager, widgetRoot) {
       return value.substr(0, idx) + newCharValue + value.substr(idx);
     }
     var newValue = simulateNewValue(newChar, currentInput);
-    var isInvalid = !valid(newValue);
+    var isInvalid = !(valid(newValue) || wouldBeValid(newValue));
```

Comment:
It does not make sense to try to identify a parametizable pattern and then access to the outer scope. Please make `wouldBeValid(newValue)` functionality be part of the `valid()` function.